### PR TITLE
Split GroupsRedux out from PhotoViewingRedux and add it to persisted whitelist

### PR DESCRIPTION
**Pull request by andrewxhill**
fixes https://github.com/textileio/photos/issues/1265
fixes https://github.com/textileio/photos/issues/1239

### CODE REVIEW

File: App/Components/GroupCard/index.tsx [hunk 1]
```diff
@@ -13,7 +13,7 @@ interface ScreenProps {
   id: string
   name: string
   members: IContact[]
-  thumb?: IFiles
+  thumb?: string
```
Comment (andrewxhill) [inline App/Components/GroupCard/index.tsx]: thumb is now just the hash of latest photo's thumb, that's all that was ever needed anyway

File: App/Containers/Cafe.tsx [hunk 2]
```diff
@@ -133,7 +133,8 @@ class Cafe extends Component<Props> {
     const date = Textile.util.timestampToDate(exp)
     const expirationDate = moment(date).format('MMMM Do YYYY [at] h:mm a')
     const refreshButtonDisabled = this.props.session.processing
-    const cafeProcessing = this.props.cafe.state === 'deregistering'
+    const cafeProcessing =
+      this.props.cafe && this.props.cafe.state === 'deregistering'
```
Comment (andrewxhill) [inline App/Containers/Cafe.tsx]: this fixes another bug, but it got me during testing so i added the fix here

File: App/Containers/Contact.tsx [hunk 3]
```diff
@@ -360,8 +357,11 @@ const mapStateToProps = (
   const adding =
     Object.keys(state.contacts.addingContacts).indexOf(address) > -1
   const directMessageThread = getDirectMessageThread(state, address)
+  const threads = getSharedThreads(state, 'name').filter(
+    thread => contact.threads.indexOf(thread.id) > -1
+  )
   return {
-    threadThumbs: getThreadThumbs(state, address, 'name'),
+    threads,
```
Comment (andrewxhill) [inline App/Containers/Contact.tsx]: no real concept of "thumbs" just the list of threads as the latest photo's thumb hash

**Comment by andrewxhill** [comment]
okay, that's GTG now. Lots of files changes so looks big, but most are basically just changed filename for import.

**Comment by andrewxhill** [comment]
oop, that is not gtg. couple more minutes here

**Comment by andrewxhill** [comment]
this is gtg. don't want to burn CI time until it's approved. will 'update branch' and get ios rerun to happen after that

**Comment by andrewxhill** [comment]
Didn't hear anything, so here she blows!

**Comment by asutula** [comment]
Ah sorry man, I did see your previous message about this not being "not g2g" so was holding off, and somehow missed your latest update. I'll give it the post-merge-review just to make sure.

**Pull request merged**